rotkehlchen/data/importer.py

```diff
@@ -1014,3 +1015,167 @@ def import_shapeshift_trades_csv(self, filepath: Path) -> Tuple[bool, str]:
+                        action_type=action_type,
+                        location=Location.UPHOLD,
+                        amount=destination_amount,
+                        asset=destination_asset,
+                        rate=None,
+                        rate_asset=None,
+                        link="",
+                        notes=notes,
+                    )
+                    self.db_ledger.add_ledger_action(action)
+            # Trades (buy)
+            else:
+                if transaction_type == 'transfer':
+                    trade = Trade(
+                        timestamp=timestamp,
+                        location=Location.UPHOLD,
+                        base_asset=destination_asset,
+                        quote_asset=origin_asset,
+                        trade_type=TradeType.BUY,
+                        amount=destination_amount,
+                        rate=Price(origin_amount / destination_amount),
+                        fee=Fee(fee),
+                        fee_currency=fee_asset,
+                        link='',
+                        notes=notes,
+                    )
+                    self.db.add_trades([trade])
+        # Outgoing
+        if origin == 'uphold' and destination != 'uphold' and transaction_type == 'out':
+            # Withdrawal
+            if destination_asset == origin_asset:
+                asset_movement = AssetMovement(
+                    location=Location.UPHOLD,
+                    category=AssetMovementCategory.WITHDRAWAL,
+                    address=None,
+                    transaction_id=None,
+                    timestamp=timestamp,
+                    asset=origin_asset,
+                    amount=origin_amount,
+                    fee=Fee(fee),
+                    fee_asset=fee_asset,
+                    link='',
+                )
+                self.db.add_asset_movements([asset_movement])
+            # Trades (sell)
+            else:
+                trade = Trade(
+                    timestamp=timestamp,
+                    location=Location.UPHOLD,
+                    base_asset=origin_asset,
+                    quote_asset=destination_asset,
+                    trade_type=TradeType.SELL,
+                    amount=origin_amount,
+                    rate=Price(destination_amount / origin_amount),
+                    fee=Fee(fee),
+                    fee_currency=fee_asset,
+                    link='',
+                    notes=notes,
+                )
+                log.debug(f'{trade.fee} {trade.rate}')
+                self.db.add_trades([trade])
+        # Incoming
+        if origin != 'uphold' and destination == 'uphold' and transaction_type == 'in':
+            # Deposit
+            if destination_asset == origin_asset:
+                asset_movement = AssetMovement(
+                    location=Location.UPHOLD,
+                    category=AssetMovementCategory.DEPOSIT,
+                    address=None,
```

Comment:
doesn't it contain info on transaction hash and address? In the UI I can see them.

Reply:
The CSV does not unfortunately and I have transactions on-chain to verify the same UI behavior.  Apparently, those details confuse accountants 🤦

Reply:
I see. Fair enough. Can't do anything then. The more info we got the better. I have had accountants and users ask for more info, especially on-chain transaction links than less.

If you got it and don't want it can just filter it. If you don't well .. you don't :smile: